scripts/inference/endpoint_generate.py

```diff
@@ -0,0 +1,206 @@
+# Copyright 2022 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Batch generate text completion results from an endpoint.
+
+Warning: This script is experimental and could change or be removed at any time
+"""
+
+import asyncio
+import copy
+import logging
+import math
+import os
+import tempfile
+import time
+from argparse import ArgumentParser, Namespace
+
+import pandas as pd
+import requests
+from composer.utils import maybe_create_object_store_from_uri, parse_uri
+
+from llmfoundry.utils import prompt_files as utils
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s %(message)s')
+log = logging.getLogger(__name__)
+
+ENDPOINT_API_KEY_ENV: str = 'ENDPOINT_API_KEY'
+ENDPOINT_URL_ENV: str = 'ENDPOINT_URL'
+
+PROMPT_DELIMITER = '\n'
+
+
+def parse_args() -> Namespace:
+    """Parse commandline arguments."""
+    parser = ArgumentParser(
+        description='Call prompts against a text completions endpoint')
+
+    #####
+    # Path Parameters
+    parser.add_argument(
```

Comment:
since this is an experimental script, im ok merging whatever version is useful to people, but would it be useful to support a remote path for the prompts file? should be pretty straightforward using composer's abstractions.